### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by macintux**
Fix and expand `README` content, update some doc strings.

One functional change: use `~ts` in `riak_cli:print` to support unicode properly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,8 +213,36 @@ this it requires status to be formatted in a specific manner when returned from
...
+
+Only `alert` values contain nested status types; e.g., a table does not contain cells which are `text` status types.
+
+## Type assembly functions
+
+See descriptions above for the arguments to each.
+
+* `riak_cli_status:text/1` - Takes an `iolist`, returns a `text` object.
+* `riak_cli_status:column/2` - Takes a title (`iolist`) and values (a list of `iolist`) intended to be displayed consecutively.
+* `riak_cli_status:table/1` - Takes a list of proplists, each representing a row in the table. The keys in the first row represent column headers; each following row (proplist) must contain the same number of taggled tuples but the keys are ignored.
```
Comment (andrewjstone) [inline README.md]: taggled -> tagged

File: README.md [hunk 2]
```diff
@@ -213,8 +213,36 @@ this it requires status to be formatted in a specific manner when returned from
...
+* `riak_cli_status:alert/1` - Takes a list of status types representing an error condition.
+
+
+## Other functions
+
+* `is_status/1` - True or false depending on whether the Erlang term argument reflects one of the 4 status types.
+* `parse/3` - a fold over a list of status objects. Takes the list, a function with two arguments (status object, accumulator) and the initial value for the accumulator as arguments.
```
Comment (andrewjstone) [inline README.md]: We should note that this function should only get called by status writers which are currently internal to riak_cli.

File: README.md [hunk 3]
```diff
@@ -213,8 +213,36 @@ this it requires status to be formatted in a specific manner when returned from
...
+
+The status api lives [here](https://github.com/basho/riak_cli/blob/develop/src/riak_cli_status.erl).
+
+## Outputting status types
+
+The status types can be converted to a string representation via [riak_cli_writer:write/1](https://github.com/basho/riak_cli/blob/develop/src/riak_cli_writer.erl) or sent directly to the console via [riak_cli:print/1](https://github.com/basho/riak_cli/blob/develop/src/riak_cli.erl)
```
Comment (andrewjstone) [inline README.md]: I don't think we should document these functions in the README. They aren't meant to be used directly. In fact I'm about to make `riak_cli:print/1` private.

File: src/riak_cli_status.erl [hunk 4]
```diff
@@ -49,9 +49,10 @@ text(IoList) ->
 column(Title, Values) ->
     {column, Title, Values}.
 
-%% @doc A table is a list of zipped lists of attribute/val pairs. Each zipped list
-%% represents a row and the first element of each pair in the first list is the
-%% title of the tables.
+%% @doc A table is constructed from a list of proplists. Each proplist
+%% represents a row in the table. The keys in the first row represent
+%% column headers; each following row (proplist) must contain the same
+%% number of taggled tuples but the keys are ignored.
```
Comment (andrewjstone) [inline src/riak_cli_status.erl]: taggled -> tagged

**Comment by macintux** [comment]
Taggled tuples, the new Erlang 18.0 hotness

**Comment by jadeallenx** [comment]
I heard Ericsson is fast tracking them for 17.5 release.

**Comment by macintux** [comment]
@andrewjstone let's talk when you're around about what we do and don't want to document in the README.

**Comment by macintux** [comment]
@andrewjstone Ready for review again

**Comment by andrewjstone** [comment]
:+1:

**Pull request merged**